File: SozlukJava/app/src/main/java/com/example/sozlukjava/model/Comment.java
package com.example.sozlukjava.model;

public class Comment {
    private int id;
    private String comment_text;
    private String created_at;
    private String username;
    private int likes;
    private int dislikes;

    public Comment(int id, String comment_text, String created_at, String username, int likes, int dislikes) {
        this.id = id;
        this.comment_text = comment_text;
        this.created_at = created_at;
        this.username = username;
        this.likes = likes;
        this.dislikes = dislikes;
    }

    public Comment() {}

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getComment_text() { return comment_text; }
    public void setComment_text(String comment_text) { this.comment_text = comment_text; }

    public String getCreated_at() { return created_at; }
    public void setCreated_at(String created_at) { this.created_at = created_at; }

    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public int getLikes() { return likes; }
    public void setLikes(int likes) { this.likes = likes; }

    public int getDislikes() { return dislikes; }
    public void setDislikes(int dislikes) { this.dislikes = dislikes; }
}

File: sozluk_flutter/lib/model/comment.dart
class Comment {
  final int id;
  final String commentText;
  final String createdAt;
  final String username;
  final int likes;
  final int dislikes;

  Comment({
    required this.id,
    required this.commentText,
    required this.createdAt,
    required this.username,
    required this.likes,
    required this.dislikes,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'],
      commentText: json['comment_text'],
      createdAt: json['created_at'],
      username: json['username'],
      likes: json['likes'],
      dislikes: json['dislikes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'comment_text': commentText,
      'created_at': createdAt,
      'username': username,
      'likes': likes,
      'dislikes': dislikes,
    };
  }
}
